storage/range.go

```diff
@@ -273,6 +274,26 @@ func (r *Range) IsLeader() bool {
 	return true
 }
 
+// canServiceCmd returns an error in the event that the range replica
+// cannot service the command as specified. This is of the case in
+// the event that the replica is not the leader.
+func (r *Range) canServiceCmd(method string, args proto.Request) error {
+	if !r.IsLeader() {
+		if !proto.IsReadOnly(method) || args.Header().ReadConsistency == proto.CONSISTENT {
+			// TODO(spencer): when we happen to know the leader, fill it in here via replica.
+			return &proto.NotLeaderError{}
+		}
+	}
+	if proto.IsReadOnly(method) {
+		if args.Header().ReadConsistency == proto.CONSENSUS {
+			return util.Errorf("consensus reads not implemented")
+		} else if args.Header().ReadConsistency == proto.INCONSISTENT && args.Header().Txn != nil {
+			return util.Errorf("cannot allow inconsistent reads within a transaction")
+		}
```

Comment:
This should also return an error if ReadConsistency is not one of the 3 defined constants (which I'm pretty sure the go protobuf implementation allows). Otherwise unknown values will be treated as `!= CONSISTENT`.